Andrew.rodriguez/dd service mapping (#18144)
DataDog/documentation: content/en/serverless/aws_lambda/configuration.md

To disable this feature, set `DD_TRACE_MANAGED_SERVICES` to `false`.

## DD_SERVICE_MAPPING

### Syntax
`DD_SERVICE_MAPPING=key1|value1,key2|value2`...
### Definition

> `DD_SERVICE_MAPPING`: This environment variable allows you to remap Datadog [Service][42] names for upstream non-Lambda (inferred spans) services. It accepts a comma-separated list of "old-service|new-service" pairs, delimited by a "|", without spaces.

Reviewer: ```suggestion
`DD_SERVICE_MAPPING`: This environment variable allows you to remap Datadog [Service][42] names for upstream non-Lambda (inferred spans) services. It accepts a comma-separated list of `old-service|new-service` pairs, delimited by a pipe (`|`), without spaces.
```

### Supported Keys for Renaming Upstream Service Names

These keys represent specific service types. If you use one of these keys in the `DD_SERVICE_MAPPING` variable, it will rename *all* upstream service names associated with that particular service type.

- `lambda_api_gateway` i.e. DD_SERVICE_MAPPING=`lambda_api_gateway|newServiceName`
- `lambda_sns` i.e. DD_SERVICE_MAPPING=`lambda_sns|newServiceName`
- `lambda_sqs` i.e. DD_SERVICE_MAPPING=`lambda_sqs|newServiceName`
- `lambda_s3` i.e. DD_SERVICE_MAPPING=`lambda_s3|newServiceName`
- `lambda_eventbridge` i.e. DD_SERVICE_MAPPING=`lambda_eventbridge|newServiceName`
- `lambda_kinesis` i.e. DD_SERVICE_MAPPING=`lambda_kinesis|newServiceName`
- `lambda_dynamodb` i.e. DD_SERVICE_MAPPING=`lambda_dynamodb|newServiceName`
- `lambda_url` i.e. DD_SERVICE_MAPPING=`lambda_url|newServiceName`

> If you wish to rename *specific* upstream services then set use these keys
> - API Gateway: use the domain i.e. `r3pmxmplak.execute-api.us-east-2.amazonaws.com|newServiceName`
> - For SNS use the topic ARN i.e. `arn:aws:sns:us-east-1:123456789012:ExampleTopic|newServiceName`
> - For SQS use the queue ARN i.e. `arn:aws:sqs:us-east-1:123456789012:MyQueue|newServiceName`
> - For S3 use the bucket ARN i.e. `arn:aws:s3:::example-bucket|newServiceName`
> - For eventbridge use the event.source i.e. `my.event|newServiceName`
> - For Kinesis use the stream ARN i.e. `arn:aws:kinesis:us-east-1:123456789012:stream/MyStream|newServiceName`
> - For DynamoDB use the table ARN i.e. `arn:aws:dynamodb:us-east-1:123456789012:table/ExampleTableWithStream|newServiceName`
> - Lambda URLS: use the domain i.e. `a8hyhsshac.lambda-url.us-east-1.amazonaws.com|newServiceName`

Reviewer: ```suggestion
If you wish to rename *specific* upstream services, use these keys:
- For API Gateway, use the domain. For example, `r3pmxmplak.execute-api.us-east-2.amazonaws.com|newServiceName`
- For SNS, use the topic ARN. For example,`arn:aws:sns:us-east-1:123456789012:ExampleTopic|newServiceName`
- For SQS, use the queue ARN. For example, `arn:aws:sqs:us-east-1:123456789012:MyQueue|newServiceName`
- For S3, use the bucket ARN. For example, `arn:aws:s3:::example-bucket|newServiceName`
- For eventbridge, use the event.source. For example, `my.event|newServiceName`
- For Kinesis, use the stream ARN. For example, `arn:aws:kinesis:us-east-1:123456789012:stream/MyStream|newServiceName`
- For DynamoDB, use the table ARN. For example, `arn:aws:dynamodb:us-east-1:123456789012:table/ExampleTableWithStream|newServiceName`
- For Lambda URLS, use the domain. For example, `a8hyhsshac.lambda-url.us-east-1.amazonaws.com|newServiceName`
```

### Examples

1. `DD_SERVICE_MAPPING="08se3mvh28.execute-api.eu-west-1.amazonaws.com:new-service-name"`  

>    This will specifically rename `08se3mvh28.execute-api.eu-west-1.amazonaws.com` to `new-service-name`, providing fine-grained control over upstream service names.

Reviewer: ```suggestion
   This specifically renames `08se3mvh28.execute-api.eu-west-1.amazonaws.com` to `new-service-name`, providing fine-grained control over upstream service names.
```

2. `DD_SERVICE_MAPPING="lambda_api_gateway:new-service-name"`  

>    This will rename all `lambda_api_gateway` services to `new-service-name`, allowing you to consolidate many services under a single Datadog Service name.

Reviewer: ```suggestion
   This renames all `lambda_api_gateway` services to `new-service-name`, allowing you to consolidate many services under a single Datadog Service name.
```

> For downstream services, please refer to `DD_SERVICE_MAPPING` in the [respective tracer's config documentation][41] for your language.

Reviewer: ```suggestion
For downstream services, refer to `DD_SERVICE_MAPPING` in the [respective tracer's config documentation][41] for your language.
```
